opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/TracingFilter.java

```diff
@@ -155,17 +167,57 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
 
             try {
                 chain.doFilter(servletRequest, servletResponse);
-                for (SpanDecorator spanDecorator: spanDecorators) {
-                    spanDecorator.onResponse(httpRequest, httpResponse, span);
+                if (!httpRequest.isAsyncStarted()) {
+                    for (SpanDecorator spanDecorator : spanDecorators) {
+                        spanDecorator.onResponse(httpRequest, httpResponse, span);
+                    }
                 }
                 // catch all exceptions (e.g. RuntimeException, ServletException...)
             } catch (Throwable ex) {
                 for (SpanDecorator spanDecorator : spanDecorators) {
-                    spanDecorator.onError(httpRequest, ex, span);
+                    spanDecorator.onError(httpRequest, httpResponse, ex, span);
                 }
                 throw ex;
             } finally {
-                span.finish();
+                if (httpRequest.isAsyncStarted()) {
+                    // what if async is already finished? This would not be called
```

Comment:
Are you going to create an issue on github repo for this, just so it doesn't get forgotten?